Invalid config value prevents app to startup with invalid stacktrace instead of validation error message

### Describe the bug

https://stackoverflow.com/questions/75535327/hibernate-validator-for-smallrye-quarkus-configmapping-throwing-exception-and-n

I faced the same problem yesterday; the problem is in io.quarkus.hibernate.validator.runtime.locale.ResteasyReactiveLocaleResolver.resolve(line 18) because a @requestcontext object is requested and request scope is not active.



### Expected behavior

Application should not start but with the right validation error message

### Actual behavior

```
Caused by: javax.validation.ValidationException: HV000149: An exception occurred during message interpolation
	at org.hibernate.validator.internal.engine.validationcontext.AbstractValidationContext.interpolate(AbstractValidationContext.java:331)
	at org.hibernate.validator.internal.engine.validationcontext.AbstractValidationContext.addConstraintFailure(AbstractValidationContext.java:231)
	at org.hibernate.validator.internal.engine.validationcontext.ReturnValueExecutableValidationContext.addConstraintFailure(ReturnValueExecutableValidationContext.java:31)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:79)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.doValidateConstraint(MetaConstraint.java:130)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.validateConstraint(MetaConstraint.java:123)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateMetaConstraint(ValidatorImpl.java:555)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateMetaConstraints(ValidatorImpl.java:537)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateReturnValueForSingleGroup(ValidatorImpl.java:1141)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateReturnValueForGroup(ValidatorImpl.java:1111)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateReturnValueInContext(ValidatorImpl.java:1033)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateReturnValue(ValidatorImpl.java:309)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateReturnValue(ValidatorImpl.java:259)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validatePropertyValue(BeanValidationConfigValidator.java:195)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validateProperty(BeanValidationConfigValidator.java:77)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validateMappingInterface(BeanValidationConfigValidator.java:57)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validateMapping(BeanValidationConfigValidator.java:39)
	at io.smallrye.config.ConfigMappings.getConfigMapping(ConfigMappings.java:124)
	at io.smallrye.config.SmallRyeConfig.getConfigMapping(SmallRyeConfig.java:411)
	at io.quarkus.arc.runtime.ConfigMappingCreator.create(ConfigMappingCreator.java:31)
	at org.acme.Config_b24e5fd41538df60276cea46d90faaf3bfa88dc9_Synthetic_Bean.create(Unknown Source)
	at org.acme.Config_b24e5fd41538df60276cea46d90faaf3bfa88dc9_Synthetic_Bean.get(Unknown Source)
	at org.acme.Config_b24e5fd41538df60276cea46d90faaf3bfa88dc9_Synthetic_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.CurrentInjectionPointProvider.get(CurrentInjectionPointProvider.java:62)
	... 80 more
Caused by: javax.enterprise.context.ContextNotActiveException: RequestScoped context was not active when trying to obtain a bean instance for a client proxy of CLASS bean [class=io.quarkus.vertx.http.runtime.CurrentVertxRequest, id=9260aa608251c44230eeb9b234f605844a7622bf]
	- you can activate the request context for a specific method using the @ActivateRequestContext interceptor binding
	at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:55)
	at io.quarkus.vertx.http.runtime.CurrentVertxRequest_ClientProxy.arc$delegate(Unknown Source)
	at io.quarkus.vertx.http.runtime.CurrentVertxRequest_ClientProxy.getCurrent(Unknown Source)
	at io.quarkus.hibernate.validator.runtime.locale.ResteasyReactiveLocaleResolver.getHeaders(ResteasyReactiveLocaleResolver.java:26)
	at io.quarkus.hibernate.validator.runtime.locale.AbstractLocaleResolver.getAcceptableLanguages(AbstractLocaleResolver.java:31)
	at io.quarkus.hibernate.validator.runtime.locale.AbstractLocaleResolver.resolve(AbstractLocaleResolver.java:17)
	at io.quarkus.hibernate.validator.runtime.locale.ResteasyReactiveLocaleResolver.resolve(ResteasyReactiveLocaleResolver.java:18)
	at io.quarkus.hibernate.validator.runtime.locale.LocaleResolversWrapper.resolve(LocaleResolversWrapper.java:27)
	at org.hibernate.validator.messageinterpolation.AbstractMessageInterpolator.interpolate(AbstractMessageInterpolator.java:343)
	at org.hibernate.validator.internal.engine.validationcontext.AbstractValidationContext.interpolate(AbstractValidationContext.java:322)
	... 103 more
```

### How to Reproduce?

https://github.com/luca-bassoricci/code-with-quarkus

build with latest version but also fails with older versions

### Output of `uname -a` or `ver`

Windows 11

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

Removing `quarkus-resteasy-reactive` will solve the problem.
I had the same problem in backend code which interacts with some `@RestClient`s during validation of PanacheEntity just before persisting, but I am unable to write a reproducer.

--- a/extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/locale/ResteasyReactiveLocaleResolver.java
+++ b/extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/locale/ResteasyReactiveLocaleResolver.java
@@ -7,6 +7,8 @@ import java.util.Map;
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
 import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
 import io.vertx.core.MultiMap;
 import io.vertx.ext.web.RoutingContext;
@@ -23,6 +25,10 @@ public class ResteasyReactiveLocaleResolver extends AbstractLocaleResolver {
 
     @Override
     protected Map<String, List<String>> getHeaders() {
+        final ManagedContext requestContext = Arc.container().requestContext();
+        if (!requestContext.isActive()) {
+            return null;
+        }
         RoutingContext current = currentVertxRequest.getCurrent();
         if (current != null) {
             Map<String, List<String>> result = new HashMap<>();

--- a/integration-tests/hibernate-validator-resteasy-reactive/src/test/java/io/quarkus/it/hibernate/validator/ConfigMappingStartupValidatorTest.java
+++ b/integration-tests/hibernate-validator-resteasy-reactive/src/test/java/io/quarkus/it/hibernate/validator/ConfigMappingStartupValidatorTest.java
@@ -1,0 +1,103 @@
+package io.quarkus.it.hibernate.validator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import jakarta.inject.Inject;
+import jakarta.validation.constraints.Pattern;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.hibernate.validator.runtime.HibernateBeanValidationConfigValidator;
+import io.quarkus.runtime.configuration.QuarkusConfigFactory;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.ConfigValidationException;
+import io.smallrye.config.ConfigValidationException.Problem;
+import io.smallrye.config.ConfigValidator;
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.SmallRyeConfigBuilder;
+import io.smallrye.config.WithDefault;
+
+@QuarkusTest
+@TestProfile(ConfigMappingStartupValidatorTest.Profile.class)
+public class ConfigMappingStartupValidatorTest {
+
+    @ConfigMapping(prefix = "config")
+    public static interface ConfigWithValidation {
+        public static final String CONFIG_WITH_VALIDATION_VALUE_MUST_BE_D_3 = "ConfigWithValidation.value() must be \"-\\d{3}\"";
+
+        @WithDefault("invalid-value")
+        @Pattern(regexp = "-\\d{3}", message = CONFIG_WITH_VALIDATION_VALUE_MUST_BE_D_3)
+        String value();
+    }
+
+    public static class Profile implements QuarkusTestProfile {
+        @Override
+        public boolean disableGlobalTestResources() {
+            return false;
+        }
+    }
+
+    private static SmallRyeConfig smallryeConfig;
+    private static Map<Class<?>, Exception> suppressedConfigValidatorExceptions;
+
+    @Inject
+    ConfigWithValidation config;
+
+    @BeforeAll
+    public static void doBefore() {
+        SmallRyeConfigBuilder builder = new SmallRyeConfigBuilder();
+        builder.withMapping(ConfigWithValidation.class).setAddDefaultSources(true)
+                .withValidator(new ConfigValidator() {
+                    final ConfigValidator base = new HibernateBeanValidationConfigValidator();
+
+                    @Override
+                    public void validateMapping(Class<?> mappingClass, String prefix, Object mappingObject)
+                            throws ConfigValidationException {
+                        try {
+                            base.validateMapping(mappingClass, prefix, mappingObject);
+                        } catch (Exception e) {
+                            suppressedConfigValidatorExceptions.put(mappingClass, e);
+                        }
+                    }
+                });
+        QuarkusConfigFactory.setConfig(smallryeConfig = builder.build());
+        Config conf = ConfigProvider.getConfig();
+        if (conf != smallryeConfig) {
+            ConfigProviderResolver cpr = ConfigProviderResolver.instance();
+            cpr.releaseConfig(conf);
+            ConfigProvider.getConfig();
+        }
+        suppressedConfigValidatorExceptions = new HashMap<>();
+    }
+
+    @AfterAll
+    public static void doAfter() {
+        ConfigProviderResolver cpr = ConfigProviderResolver.instance();
+        cpr.releaseConfig(smallryeConfig);
+        smallryeConfig = null;
+        suppressedConfigValidatorExceptions = null;
+    }
+
+    @Test
+    public void test() {
+        assertEquals("invalid-value", config.value());
+        ConfigValidationException ex = assertThrows(ConfigValidationException.class, () -> {
+            throw suppressedConfigValidatorExceptions.get(ConfigWithValidation.class);
+        });
+        assertEquals(1, ex.getProblemCount());
+        final Problem problem = ex.getProblem(0);
+        assertEquals("config.value " + ConfigWithValidation.CONFIG_WITH_VALIDATION_VALUE_MUST_BE_D_3, problem.getMessage());
+    }
+}

